src/parser/statement.js

```diff
@@ -881,10 +884,26 @@ export default class StatementParser extends ExpressionParser {
         if (this.hasPlugin("decorators2")) {
           this.resetStartLocationFromNode(member, decorators[0]);
         }
-        decorators = [];
       }
 
       this.parseClassMember(classBody, member, state);
+      const isClassField = member.kind == null;
+
+      if (isClassField) {
+        if (this.eat(tt.comma)) {
+          this.state.inCommaSeparatedFields = true;
+          // Copy decorators for cases like `@foo x, y;`
+          if (decorators.length) {
+            member.decorators = decorators;
```

Comment:
Let's slice here. It'll be really weird if a transform pushes a decorator and two fields get it.